### PR TITLE
add quill version 4.3.0

### PR DESCRIPTION
**Pull request by odygrd**
`quill` is a C++17 low latency logging library.  The project can be found [here](https://github.com/odygrd/quill)

### CODE REVIEW

**Comment by bazel-io** [comment]
Hello @bazelbuild/bcr-maintainers, modules without existing maintainers (quill) have been updated in this PR. Please review the changes.

**Comment by odygrd** [comment]
Please add the presubmit-auto-run and skip-url-stability-check labels to the PR. Thanks!

File: modules/quill/4.3.0/patches/add_build_file.patch [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
++           hdrs = glob(["quill/include/**/*.h"]),
++           includes = ["quill/include"],
++           linkopts = select({
++               "@platforms//os:windows": [],
++               "@platforms//os:linux": ["-lpthread", "-lrt",],
++               "//conditions:default": ["-lpthread",]
++           }),
++           copts = select({
++               "@platforms//os:windows": [],
++               "//conditions:default": ["Wno-gnu-zero-variadic-macro-arguments"],
```
Comment (fmeum) [inline modules/quill/4.3.0/patches/add_build_file.patch:18]: This looks like a `gcc`-specific argument. You could select on `@rules_cc//cc/compiler:gcc` instead.

**Comment by odygrd** [comment]
any advice on how to fix the CI builds please ?

**Comment by meteorcloudy** [comment]
You need to add platforms as a direct dependency, I pushed a new commit to do that.

**Comment by meteorcloudy** [comment]
Can you please update the MODULE.bazel file patch and its integrity value?

**Comment by odygrd** [comment]
thank you @meteorcloudy !

**Review by meteorcloudy — APPROVED** [review]

**Comment by meteorcloudy** [comment]
Thanks for the contribution! Do you mind also upstreaming the BUILD and MODULE.bazel file so that we don't need those patches for future version?

**Comment by odygrd** [comment]
> Thanks for the contribution! Do you mind also upstreaming the BUILD and MODULE.bazel file so that we don't need those patches for future version?

I am planning to upstream them. I was thinking of placing them within a `/bazel` directory, following the pattern used by libfmt [here](https://github.com/fmtlib/fmt/tree/master/support/bazel). 

Does that align with what you had in mind, or would you prefer them at the root directory? I'm flexible either way.

**Comment by meteorcloudy** [comment]
> To get [Bazel](https://bazel.build/) working with {fmt} you can copy the files BUILD.bazel, MODULE.bazel, WORKSPACE.bazel, and .bazelversion from this folder (support/bazel) to the root folder of this project.

If it's in a subfolder, users still need to copy it out to make the Bazel build work, which means BCR module still needs a patch file. So if you don't mind, it's better to add those files where Bazel expects them to be.

**Pull request merged**